### PR TITLE
Fix setup hook order [wip]

### PR DESCRIPTION
**Pull request by SeanNaren**
## What does this PR do?

Fixes call ordering for set up hook based on release/1.2-dev.

Here we see setup being called on the accelerator:

https://github.com/PyTorchLightning/pytorch-lightning/blob/release/1.2-dev/pytorch_lightning/trainer/trainer.py#L504

which internally calls the setup trainer hook:

https://github.com/PyTorchLightning/pytorch-lightning/blob/7f8fdda9a2c43e679e29fcab17b7a9e428367c55/pytorch_lightning/accelerators/legacy/gpu_accelerator.py#L44

Since we'd like to move this trainer hook to the trainer, we need to ensure we call it before calling the accelerator setup to imitate the behaviour defined already.

### CODE REVIEW

File: pytorch_lightning/trainer/trainer.py [hunk 1]
```diff
@@ -473,8 +474,6 @@ def fit(
         self.training_type_plugin.pre_training()
         self.precision_plugin.pre_training()
 
-        self.call_setup_hook(self.lightning_module)
```
Comment (tchaton) [inline pytorch_lightning/trainer/trainer.py:476]: Please, move `on_before_accelerator_backend_setup ` before `self.accelerator_backend.setup`. It is a bug.

**Review by tchaton — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Review by Borda — APPROVED** [review]
mind add it in chlog?

**Comment by SeanNaren** [comment]
[DO NOT MERGE] Will do, for full transparency we're waiting on the `accelerator-refactor-sharded` branch to test green, and then we can evaluate this PR. Changing hook order will probably cause a few tests to fail

**Comment by kaushikb11** [comment]
> [DO NOT MERGE] Will do, for full transparency we're waiting on the `accelerator-refactor-sharded` branch to test green, and then we can evaluate this PR. Changing hook order will probably cause a few tests to fail

@SeanNaren Could turn it into Draft then?

**Comment by awaelchli** [comment]
@SeanNaren The tests in 
`tests/callbacks/test_callbacks.py::test_trainer_callback_system 
`needs to be adapted for this change.
otherwise looks good, all other tests pass for me.

**Review by awaelchli — APPROVED** [review]

**Pull request merged**